src/curl_internal.hpp

```diff
@@ -0,0 +1,43 @@
+#ifndef POWERLOADER_SRC_CURL_INTERNAL_HPP
+#define POWERLOADER_SRC_CURL_INTERNAL_HPP
+
+#include <powerloader/curl.hpp>
+
+namespace powerloader::details
+{
+    // Scoped initialization and termination of CURL.
+    // This should never have more than one instance live at any time.
```

Comment:
This should be enforced by deleting copy constructor and assign operator, and implementing a proper ref count in the constructor (and deciding th policy in case someone tries to instantiate it twice).

Reply:
I didnt bother adding that because it's now an implementation detail of the Context which already have these features, but I will see what I can do. I do need it to be at least move-enabled. Not a ref count but an atomic check in the constructor yes, like in Context.